src/provider/types.ts

```diff
@@ -8,9 +8,15 @@ import type {
   SignMultipleTransactionsResponse,
   SignTransactionResponse,
 } from '../transactions';
+import { RpcResponse } from '../types';
 
 interface BaseBitcoinProvider {
-  call: (request: string) => Promise<CallWalletResponse>;
+  request: <Method extends keyof Requests>(
+    method: Method,
+    options: Params<Method>,
+    providerId?: string
+  ) => Promise<RpcResponse<Method>>;
+  listen: (method: string, callback: () => void) => void;
```

Comment:
Just another note here that this should be strongly typed; the method and the args that the callback will get.